stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/internal/Linkify.kt

```diff
@@ -0,0 +1,212 @@
+package io.getstream.chat.android.ui.common.internal
+
+import android.annotation.SuppressLint
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.Spanned
+import android.text.method.LinkMovementMethod
+import android.text.style.URLSpan
+import android.text.util.Linkify
+import android.widget.TextView
+import androidx.core.util.PatternsCompat
+import java.util.ArrayList
+import java.util.Collections
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+/**
+ * Linkify links the part of the text based on matching pattern.
+ *
+ * This class is a simplified version of [Linkify] and differs only in one following way
+ * It doesn't remove any existing URLSpan from the Spannable.
+ */
+internal object Linkify {
+
+    private val COMPARATOR: Comparator<LinkSpec> = object : Comparator<LinkSpec> {
+        override fun compare(a: LinkSpec, b: LinkSpec): Int {
+            if (a.start < b.start) {
+                return -1
+            }
+            if (a.start > b.start) {
+                return 1
+            }
+            if (a.end < b.end) {
+                return 1
+            }
+            return if (a.end > b.end) {
+                -1
+            } else 0
+        }
+    }
+
+    /**
+     * Scans the provided TextView and turns all occurrences
+     * of the link types into clickable links (Currently only support web urls).
+     * If matches are found the movement method for the TextView is set to
+     * LinkMovementMethod.
+     *
+     * NOTE: Because this implementation doesn't remove existing URLSpan,
+     * make sure it is not repeatedly called on same text.
+     *
+     * @param textView TextView whose text is to be marked-up with links.
+     */
+    internal fun addLinks(textView: TextView) {
+        val t: CharSequence = textView.text
+
+        if (t is Spannable) {
+            if (addLinks(t)) {
+                addLinkMovementMethod(textView)
+            }
+        } else {
+            val s = SpannableString.valueOf(t)
+            if (addLinks(s)) {
+                addLinkMovementMethod(textView)
+                textView.text = s
+            }
+        }
+    }
+
+    /**
+     * Scans the provided spannable text and turns all occurrences
+     * of the link types into clickable links (Currently only support web urls).
+     *
+     * @param text Spannable whose text is to be marked-up with links.
+     * @return True if at least one link is found and applied.
+     */
+    @SuppressLint("RestrictedApi")
+    private fun addLinks(text: Spannable): Boolean {
+        val links: ArrayList<LinkSpec> = ArrayList<LinkSpec>()
```

Comment:
Normally we use `listOf<T>()` and interfaces instead of concrete implementations (List instead of ArrayList).